### PR TITLE
LiteratureBuilder: fail on duplicated docs and figures

### PR DESCRIPTION
**Pull request by david-caro**
Signed-off-by: David Caro <david@dcaro.es>

### CODE REVIEW

File: inspire_schemas/builders/literature.py [hunk 1]
```diff
@@ -909,8 +917,13 @@ def add_figure(
             if locals()[dict_key] is not None:
                 figure[dict_key] = locals()[dict_key]
 
-        if figure:
-            self._append_to('figures', figure)
+        if key_already_there(figure, self.record.get('figures', ())):
+            raise ValueError(
+                'There\'s already a figure with the key %s.'
```
Comment (michamos) [inline inspire_schemas/builders/literature.py]: Is the `key` never unicode?

File: inspire_schemas/builders/literature.py [hunk 2]
```diff
@@ -90,6 +90,14 @@ def _item_has_page_or_artid(item):
     return has_pub_info and has_page_or_artid
 
 
+def key_already_there(element, elements):
+    """Checks if the given element existng by using the key `key`."""
+    for existing_element in elements:
```
Comment (michamos) [inline inspire_schemas/builders/literature.py]: This is making insertion quadratic in the number of elements. Maybe better to maintain two `set`s of keys on the side.

File: inspire_schemas/builders/literature.py (hunk 1 above)
Comment (david-caro) [reply inspire_schemas/builders/literature.py]: It does not matter:
```python
In [1]: lolo = u'sé'

In [2]: 'bom %s' % lolo
Out[2]: u'bom s\xe9'
```

File: inspire_schemas/builders/literature.py (hunk 2 above)
Comment (david-caro) [reply inspire_schemas/builders/literature.py]: I don't really think we should care much about that now, we will have ~100 figures/documents, that's peanuts, imo not really worth the complexity of adding the sets.

**Pull request merged**